sdks/python/apache_beam/runners/worker/statesampler.py

```diff
@@ -0,0 +1,99 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This module is experimental. No backwards-compatibility guarantees.
+import threading
+from collections import namedtuple
+
+from apache_beam.utils.counters import Counter
+from apache_beam.utils.counters import CounterName
+
+try:
+  from apache_beam.runners.worker import statesampler_fast as statesampler_impl
+  FAST_SAMPLER = True
+except ImportError:
+  from apache_beam.runners.worker import statesampler_slow as statesampler_impl
+  FAST_SAMPLER = False
+
+
+class ExecutionStateSamplers(threading.local):
+  """ Per-thread state sampler. """
+  def __init__(self):
+    super(ExecutionStateSamplers, self).__init__()
+    self._current_sampler = None
+
+  def current_sampler(self):
+    return self._current_sampler
+
+  def set_sampler(self, sampler):
+    self._current_sampler = sampler
+
+
+EXECUTION_STATE_SAMPLERS = ExecutionStateSamplers()
```

Comment:
<!--thread_id:cc_162192161_t; commit:4c27aa0fbbbc31e24e3b8d6bc04128b44d18b9ec; resolved:0-->
<!--section:context-quote-->
> **robertwb** wrote:
> Please get rid of this unused global state.

<!--section:body-->
I plan to use the global state to track metrics and logging (and eventually remove the old style metrics context). I can add this now, or on the next PR. What do you think?
Next PR: https://github.com/apache/beam/pull/4387/files#diff-89072ff532dd15a0af899957de4c26f3R152